lib/cfg-lexer.c

```diff
@@ -1069,6 +1047,52 @@ relex:;
       self->non_pragma_seen = TRUE;
     }
 
+  return TRUE;
+}
+
+int
+cfg_lexer_lex(CfgLexer *self, YYSTYPE *yylval, YYLTYPE *yylloc)
+{
+  /*
+   * NOTES:
+   *
+   * 1) the role of the relex label is to restart the lexing process once
+   * new tokens were injected into the input stream.  (e.g.  after a
+   * generator was called).  This should really be a loop, and quite
+   * possible any refactors should start here by eliminating that
+   * loop-using-goto
+   *
+   * 2) make note that string tokens are allocated by malloc/free and not
+   * g_malloc/g_free, this is significant.  The grammar contains the free()
+   * call, so getting rid of that would require a lot of changes to the
+   * grammar. (on Windows glib, malloc/g_malloc are NOT equivalent)
+   *
+   */
+
+  gint tok;
+  gboolean relex_required;
+
+relex:;
+
+  gboolean is_token_injected = cfg_lexer_consume_next_injected_token(self, &tok, yylval, yylloc);
+
+  if (!is_token_injected)
+    {
+      if (cfg_lexer_get_context_type(self) == LL_CONTEXT_BLOCK_CONTENT)
+        cfg_lexer_start_block_state(self, "{}");
+      else if (cfg_lexer_get_context_type(self) == LL_CONTEXT_BLOCK_ARG)
+        cfg_lexer_start_block_state(self, "()");
+
+      tok = cfg_lexer_lex_next_token(self, yylval, yylloc);
+      cfg_lexer_append_preprocessed_output(self, self->token_pretext->str);
+    }
+
+  if (!cfg_lexer_process_token(self, tok, yylval, yylloc, &relex_required))
```

Comment:
This function is a little confusing for me. It has a return value, and other bool value via parameter (`relex_required`), and those two variable describes what the next action should be. Those two variable are tangled, but the code does not represents it.
Please if possible couple those variables.